src/index.ts

```diff
@@ -288,6 +382,109 @@ function createRdfEmitter(program: Program) {
     });
   }
 
+  function checkIfDecoratorsGeneric(prop: ModelProperty) {
+    let setDecorators = new Set();
+    for (let dec of prop.decorators) {
+      {
+        setDecorators.add(dec.decorator.name);
+      }
+    }
+
+    setDecorators.delete("$summary");
+    setDecorators.delete("$doc");
+    setDecorators.delete("$deprecated");
+
+    if (setDecorators.size == 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function writeDecoratorsInBuiltGeneralInfo(
+    program: Program,
+    m: Model | ModelProperty,
+    object: NamedNode,
+    arrayQuads: Quad[]
+  ) {
+    const doc = getDoc(program, m);
+    if (doc) {
+      arrayQuads.push(
+        quad(object, nn("rdfs:comment"), DataFactory.literal(doc))
+      );
+    }
+
+    const summary = getSummary(program, m);
+    if (summary) {
+      arrayQuads.push(
+        quad(object, nn("sh:summary"), DataFactory.literal(summary))
+      );
+    }
+
+    const deprecated = getDeprecated(program, m);
+    if (deprecated) {
+      arrayQuads.push(
+        quad(object, nn("sh:deprecated"), DataFactory.literal(deprecated))
+      );
+    }
+  }
+
+  function writeDecoratorsInBuiltConstraints(
+    program: Program,
+    m: Model | ModelProperty,
+    object: NamedNode,
+    arrayQuads: Quad[]
+  ) {
+    const maxLength = getMaxLength(program, m);
+    if (maxLength) {
+      arrayQuads.push(
+        quad(object, nn("sh:maxLength"), DataFactory.literal(maxLength))
+      );
+    }
+
+    const minLength = getMinLength(program, m);
+    if (minLength) {
+      arrayQuads.push(
+        quad(object, nn("sh:minLength"), DataFactory.literal(minLength))
+      );
+    }
+
+    const minValue = getMinValue(program, m);
+    if (minValue) {
+      arrayQuads.push(
+        quad(object, nn("sh:minValue"), DataFactory.literal(minValue))

```

Comment:
I think minValue and maxValue are `minInclusive` and `maxInclusive` respectively.